### DISCUSSION TITLE
Why should I choose Coolify over Caprover or Dokku?

### DISCUSSION THREAD

**Discussion by kamzata** [coollabsio/coolify#688] [Q&A]
They seems all great projects. Coolify seems more recent than Caprover and Dokku. Which one should I choose and why?

**Comment by lcfd** [comment]
Hi! 👋 

I tried this project some months ago.
Even then he was very well looked after and incredibly active in development.
From what I can see it still is and getting better and better.
Dokku and Caprover seemed much rawer and more developer-friendly to me. Here you will find a good UX and a lot of comfortable things.

At the moment I don't use it anymore because I don't need a lot of services and a simple docker-compose is enough.
I'm still monitoring the development because I would like to use it again in the future.

**Reply by kamzata** [reply]
Dokku seems great if you don't need a GUI. Last Caprover release is dated 10 October 2021 and it seems developed mainly by just one person as for Coolify even though this latest is a more recent project. They both use Apache licence. However, now I discovered also [Umbrel](https://umbrel.com/).

**Reply by lcfd** [reply]
Yes, Caprover looks a bit outdated and when I tried to use it it wasn't a 100% good experience.

Talking about Umbrel I don't like the license (PolyForm Noncommercial License 1.0.0 ???). Not really open source.
The rest looks well done but I haven't tried it because of the license.

**Reply by kamzata** [reply]
> Yes, Caprover looks a bit outdated and when I tried to use it it wasn't a 100% good experience.
> 
> Talking about Umbrel I don't like the license (PolyForm Noncommercial License 1.0.0 ???). Not really open source. The rest looks well done but I haven't tried it because of the license.

So... the choice is between Caprover and Coolify... however, Caprover supports clusters and private docker registry. Does Coolify?

**Reply by fd6130** [reply]
Coolify only support public repository? Mean I cannot deploy my private work project then :/

**Reply by FrancocDev** [reply]
> Coolify only support public repository? Mean I cannot deploy my private work project then :/

You can use private repos. Just Create New Resource > Git Source and then when you add a project, at the top, you can choose Git and search in your repos.

**Reply by sanfx** [reply]
I hope you guys settled on something by now or not?

**Reply by code-flag** [reply]
Still waiting for someone to tell us their experience with caprover and coolify. Currently I'm feeling I should choose caprover due to many other things promised by it especially advance scalability and load balancer

**Reply by peaklabs-dev** [reply]
Scalability with Kubernetes is coming soon as soon as v4 hits stabel we are just finishing up the last bugs and then it will arrive.
